### PR TITLE
Add redactedKeys for removing sensitive values from metadata

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal
Adds the `redactedKeys` property to `Configuration` which allows a user to specify sensitive keys. Any sensitive values present in the metadata of `BugsnagEvent` will be replaced by `"[REDACTED]"` when serialized to JSON.

## Changeset
- Added `redactedKeys` property to `Configuration`, which by default is an array of "password"
- Set the `redactedKeys` on each `BugsnagEvent` using the `Configuration` value
- Sanitised the metadata at the point of JSON serialization when 
- `BugsnagEvent` now redacts sensitive values with any key that is considered redacted

## Tests
- Added unit tests to verify the default value of `redactedKeys` and its effect on JSON serialization
- Added tests to macOS/tvOS projects which already existed on the file system but which were not part of the xcodeproj file

### CODE REVIEW

File: Source/BugsnagEvent.m [hunk 1]
```diff
@@ -308,6 +308,8 @@ - (BOOL)shouldBeSent;
  *  The release stage of the application
  */
 @property(readwrite, copy, nullable) NSString *releaseStage;
+
+@property NSArray *redactedKeys;
```
Comment (robinmacharg) [inline Source/BugsnagEvent.m:312]: I think the comment should point to the Config comment.

File: Source/BugsnagConfiguration.h [hunk 2]
```diff
@@ -107,6 +107,17 @@ typedef NS_OPTIONS(NSUInteger, BSGErrorType) {
...
  */
 @property(readwrite, retain, nullable) NSArray *enabledReleaseStages;
+
+/**
+ * Sets which values should be removed from any Metadata objects before
+ * sending them to Bugsnag. Use this if you want to ensure you don't send
+ * sensitive data such as passwords, and credit card numbers to our
+ * servers. Any keys which contain these strings will be filtered.
+ *
+ * By default, redactedKeys is set to "password"
```
Comment (robinmacharg) [inline Source/BugsnagConfiguration.h:117]: nit: the default value is `[@"password"]`

File: Source/BugsnagConfiguration.h [hunk 3]
```diff
@@ -107,6 +107,17 @@ typedef NS_OPTIONS(NSUInteger, BSGErrorType) {
...
+ */
+@property(readwrite, retain, nullable) NSArray<NSString *> *redactedKeys;
```
Comment (robinmacharg) [inline Source/BugsnagConfiguration.h:119]: This will need to be an array of ids to support RegEx matching.

File: Tests/BugsnagSinkTests.m [hunk 4]
```diff
@@ -20,7 +20,7 @@ @interface BugsnagSinkTests : XCTestCase
 @end
 
 @interface BugsnagSink ()
-- (NSDictionary *)getBodyFromReports:(NSArray *)reports;
+- (NSDictionary *)getBodyFromReports:(NSArray *)events;
```
Comment (robinmacharg) [inline Tests/BugsnagSinkTests.m:23]: This method is only mentioned here and in the test, a total of 6 times, AFAICT.  I'd be happy for it to be renamed to `getBodyFromEvents:` as part of this change; I think it's clear enough housekeeping.

File: Source/BugsnagConfiguration.h (hunk 2 above)
Comment (fractalwrench) [reply Source/BugsnagConfiguration.h:117]: Updated docs to reflect this.

File: Tests/BugsnagSinkTests.m (hunk 4 above)
Comment (fractalwrench) [reply Tests/BugsnagSinkTests.m:23]: Agreed - renamed this method.

File: Source/BugsnagConfiguration.h (hunk 3 above)
Comment (fractalwrench) [reply Source/BugsnagConfiguration.h]: Updated implementation to support regex testing, added an additional test case.

**Pull request merged**